src/System.Private.ServiceModel/src/System/IdentityModel/Claims/WindowsClaimSet.cs

```diff
@@ -1,16 +1,349 @@
+        }
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _disposed = true;
+                _windowsIdentity.Dispose();
+            }
+        }
+
+        IList<Claim> InitializeClaimsCore()
+        {
+            if (_windowsIdentity.AccessToken == null)
+                return new List<Claim>();
+
+            List<Claim> claims = new List<Claim>(3);
+            claims.Add(new Claim(ClaimTypes.Sid, _windowsIdentity.User, Rights.Identity));
+            Claim claim;
+            if (TryCreateWindowsSidClaim(_windowsIdentity, out claim))
+            {
+                claims.Add(claim);
+            }
+            claims.Add(Claim.CreateNameClaim(_windowsIdentity.Name));
+            if (_includeWindowsGroups)
+            {
+                // claims.AddRange(Groups);
+            }
+            return claims;
+        }
+
+        void EnsureClaims()
+        {
+            if (_claims != null)
+                return;
+
+            _claims = InitializeClaimsCore();
+        }
+
+        void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ObjectDisposedException(this.GetType().FullName));
+            }
+        }
+
+        static bool SupportedClaimType(string claimType)
+        {
+            return claimType == null ||
+                ClaimTypes.Sid == claimType ||
+                ClaimTypes.DenyOnlySid == claimType ||
+                ClaimTypes.Name == claimType;
+        }
+
+        // Note: null string represents any.
+        public override IEnumerable<Claim> FindClaims(string claimType, string right)
+        {
+            ThrowIfDisposed();
+            if (!SupportedClaimType(claimType) || !ClaimSet.SupportedRight(right))
+            {
+                yield break;
+            }
+            else if (_claims == null && (ClaimTypes.Sid == claimType || ClaimTypes.DenyOnlySid == claimType))
+            {
+                if (ClaimTypes.Sid == claimType)
+                {
+                    if (right == null || Rights.Identity == right)
+                    {
+                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity.User, Rights.Identity);
+                    }
+                }
+
+                if (right == null || Rights.PossessProperty == right)
+                {
+                    Claim sid;
+                    if (TryCreateWindowsSidClaim(_windowsIdentity, out sid))
+                    {
+                        if (claimType == sid.ClaimType)
+                        {
+                            yield return sid;
+                        }
+                    }
+                }
+
+                if (_includeWindowsGroups && (right == null || Rights.PossessProperty == right))
+                {
+                    // Not sure yet if GroupSidClaimCollections are necessary in .NET Core
```

Comment:
Remove commented code. If you want to catch this code path being hit, keep the if clause and throw.